webcompat/static/js/lib/issue-list.js

```diff
@@ -191,42 +188,53 @@ issueList.SearchView = Backbone.View.extend({
   updateSearchQuery: function(data) {
     this.input.val(data);
   },
-  _isEmpty: true,
-  _currentSearch: null,
-  searchIfNotEmpty: _.debounce(function(e) {
+  doSearch: _.debounce(function(e) {
     var searchValue = e.type === 'click' ? $(e.target).prev().val() :
                       e.target.value;
-    if (searchValue.length) {
-      this._isEmpty = false;
-      // don't search if nothing has changed
-      // (or user just added whitespace)
-      if ($.trim(searchValue) !== this._currentSearch) {
-        this._currentSearch = $.trim(searchValue);
-        this.doSearch(this._currentSearch);
-        // clear any filters that have been set.
-        issueList.events.trigger('filter:clear', {removeQ: false});
-      }
-    }
+    this.mainView.params.setParam('search', searchValue);
+  }, 250),
+});
```

Comment:
I guess we don't prevent doing search for the same thing, or an empty search now. Is that intentional? (maybe you do later on, I haven't read everything yet).